packages/astro/src/core/app/index.ts

```diff
@@ -271,28 +271,27 @@ export class App {
 			}
 			response = await this.#pipeline.renderRoute(renderContext, pageModule);
 		} catch (err: any) {
-			if (err instanceof EndpointNotFoundError) {
-				return this.#renderError(request, { status: 404, response: err.originalResponse });
-			} else {
-				this.#logger.error(null, err.stack || err.message || String(err));
-				return this.#renderError(request, { status: 500 });
-			}
+			this.#logger.error(null, err.stack || err.message || String(err));
+			return this.#renderError(request, { status: 500 });
 		}
 
-		// endpoints do not participate in implicit rerouting
-		if (routeData.type === 'page' || routeData.type === 'redirect') {
-			if (REROUTABLE_STATUS_CODES.has(response.status)) {
-				return this.#renderError(request, {
-					response,
-					status: response.status as 404 | 500,
-				});
-			}
+		if (REROUTABLE_STATUS_CODES.has(response.status) && response.headers.get("X-Astro-Reroute") !== "no") {
+			return this.#renderError(request, {
```

Comment:
Why did we remove the `routeData.type === 'page' || routeData.type === 'redirect'` check?

Reply:
That was there to carve out endpoints from the rerouting behavior. That does not happen anymore.